Interface.C

```diff
@@ -134,27 +139,80 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
             }
         }
 
+
         // Pass the mesh vertices information to preCICE
-        precice_.setMeshVertices(meshID_, numDataLocations_, vertices, vertexIDs_);
+        precice_.setMeshVertices(meshID_, numDataLocations_, vertices, centerIDs_);
+
+
+        //Only set the triangles, when it is necessary
+        if (provideMeshConnectivity_)
+        {
+
+            for (uint j = 0; j < patchIDs_.size(); j++)
+            {
+
+                //Define triangles
+
+                const List<face> faceField=mesh.boundaryMesh()[patchIDs_.at(j)].localFaces();
+
+                const Field<point> pointCoords= mesh.boundaryMesh()[patchIDs_.at(j)].localPoints();
+
+                //Array to transform coords in precice IDs
+                double triCoords[faceField.size()*18];
+
+                unsigned int coordIndex=0;
+
+                forAll(faceField,facei){
+
+                    const face& quad_face=faceField[facei];
+
+                    faceTriangulation faceTri(pointCoords,quad_face,false);
+
+                    for(uint triIndex=0; triIndex<2; triIndex++){
+                        for(uint nodeIndex=0; nodeIndex<3; nodeIndex++){
+                            for(uint xyz=0; xyz<3; xyz++)
+                                triCoords[coordIndex++]=pointCoords[faceTri[triIndex][nodeIndex]][xyz];
+                        }
+                    }
+                }
+
+
+                //Number of precice IDs
+                int triVertIDs[faceField.size()*6];
+
+                //Get precice IDs
+                precice_.getMeshVertexIDsFromPositions(meshID_,faceField.size()*6,triCoords,triVertIDs);
+
+
+                Info<<"Number of Triangles to set "<<faceField.size()*2<<endl;
+
+                //Set Triangles
+                for(int faceI=0; faceI<faceField.size()*2; faceI++){
+                    precice_.setMeshTriangleWithEdges(meshID_,triVertIDs[faceI*3],triVertIDs[faceI*3+1], triVertIDs[faceI*3+2]);
+                }
+            }
+        }
+
     }
-    else
+    if (!(locationsType_ == "faceNodes" || locationsType_ == "faceCenters" || locationsType_ == "faceCentres") )
     {
         FatalErrorInFunction
-             << "ERROR: interface points location type "
-             << locationsType_
-             << " is invalid."
-             << exit(FatalError);
+                << "ERROR: interface points location type "
+                << locationsType_
+                << " is invalid."
+                << exit(FatalError);
     }
 }
 
 
+//the readers and writers distinguish, where to use
```

Comment:
In fact, we relate now the connectivity to a patch. Hence, the writer/reader needs to know, what to use for which patch. This is independent of the connectivity/triangles. In my opinion, the comment is a little bit misleading, if we argue here with triangles.

Reply:
I am a bit confused... Why do we relate the connectivity to a patch? I see three levels: adapter-wide, per interface, and per patch. I recommend we relate to the interface, if possible.

But still, let's try keep the comments as descriptive as possible, so that our unknown future reader will also understand. :)

Reply:
Sorry, this was my fault, it is connected to the interface. I will add a more detailed comment on this.